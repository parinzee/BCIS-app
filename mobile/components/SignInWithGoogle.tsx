import * as React from "react";
import * as Linking from "expo-linking";
import { StyleSheet, Image, Text, TouchableOpacity, View } from "react-native";
import { ActivityIndicator, Colors } from "react-native-paper";
import { URLConfiguration } from "../constants/AWSCognito";
import { handleGoogleCognitoCallback } from "../constants/AWSCognito";

export default function SignInWithGoogle() {
  const [isLoading, setIsLoading] = React.useState(false);

  React.useEffect(() => {
    const subscription = Linking.addEventListener("url", async (event) => {
      setIsLoading(true);
      await handleGoogleCognitoCallback(event);
      setIsLoading(false);
    });

    return () => {
      subscription.remove();
    };
  });

  return (
    <View style={{ ...styles.container, marginTop: 10 }}>
      <TouchableOpacity
        style={styles.container}
        onPress={() => Linking.openURL(URLConfiguration.googleURL)}
        disabled={isLoading}
      >
        <Image
          source={require("../assets/images/g-normal.png")}
          style={styles.logo}
        />
        {!isLoading ? (
          <Text style={styles.text}>Sign In With Google</Text>
        ) : (
          <ActivityIndicator />
        )}
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    height: 48,
    width: 230,
    backgroundColor: Colors.white,
    alignSelf: "center",
    borderRadius: 10,
    overflow: "hidden",
    flexDirection: "row",
    alignItems: "center",
  },
  logo: {
    height: 46,
    width: 46,
  },
  text: {
    marginLeft: 5,
    fontSize: 15,
    color: Colors.black,
    fontWeight: "bold",
  },
});
