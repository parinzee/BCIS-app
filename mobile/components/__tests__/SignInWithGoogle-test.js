import * as React from "react";
import renderer from "react-test-renderer";
import SignInWithGoogle from "../SignInWithGoogle";

it(`renders correctly`, () => {
  const tree = renderer.create(<SignInWithGoogle />).toJSON();

  expect(tree).toMatchSnapshot();
});
