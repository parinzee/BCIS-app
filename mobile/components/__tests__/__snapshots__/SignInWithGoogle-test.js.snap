// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 10,
      "flexDirection": "row",
      "height": 48,
      "marginTop": 10,
      "overflow": "hidden",
      "width": 230,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 48,
        "opacity": 1,
        "overflow": "hidden",
        "width": 230,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 46,
          "width": 46,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 15,
          "fontWeight": "bold",
          "marginLeft": 5,
        }
      }
    >
      Sign In With Google
    </Text>
  </View>
</View>
`;
